### PR TITLE
[MO][POC] TensorBoard for ov::Model - Alternative to Netron

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Implemented alternative way to visualize IR, ov::Model via TensorBoard tool.
Implemented functionality generates TensorBoard logs for IR and ov::Model

Currently the frameworks (PyTorch, Paddle, not only TensorFlow) use TensorBoard to visualize their models. The idea is to generate TensorBoard logs for the input model and open it up via TensorBoard tool.
Here is an example for PyTorch by this [link](https://pytorch.org/tutorials/intermediate/tensorboard_tutorial.html).

There is a similar way to visualize Pytorch model in TensorBoard like as follows:

```
from torch.utils.tensorboard import SummaryWriter
writer = SummaryWriter('logs_dir')
writer.add_graph(net, images)
writer.close()
```

In OpenVINO we have two ways to generate TB logs. The first one is using OpenVino Tools API:

```
from openvino.tools.mo.utils.tensorboard.writer import SummaryWriter
writer = SummaryWriter('logs_dir')
writer.add_graph(ov_model)
writer.close()
```

The second way is using new MO option `--tensorboard_ir_logdir`.

Example of how IR for FaceNet is displayed in TensorBoard:
image

TODO:

* [ ] Support Loop, If
* [ ] Support attributes
* [ ] Constant values
* [ ] Fill GitHub issue in TB repo: complain on bugs in the visualization

**Ticket:** TBD

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

### CODE REVIEW

**Comment by ilyachur** [comment]
This approach will work only wit MO.
Why do we need a new visualisation tool for IR? Can we contribute to `Netron` if we have any problems?

**Comment by rkazants** [comment]
> This approach will work only wit MO. Why do we need a new visualisation tool for IR? Can we contribute to `Netron` if we have any problems?

Ilya, good question. Currently,  we have a problem with Netron to visualize graphs with Loops and Ifs operations because they have body graphs, etc. It is not easy to add such support to there and we should ask the main contributors to wait for this feature. Several years passed but it is still unimplemented. In the meantime, we can add it this feature very easy, it costs just couple of hours:) Really easy to maintain from our side in the centralized way.

**Comment by slyalin** [comment]
The _only_ problem with TB is that you must prepare a model to be visualized by writing some code. So even if you already have a model represented as a file (IR -- in out case) it is not enough to look at it. If you have the model in Python you still need to serialize it twice -- for IR itself and to format that TB can accept. This is a major and probably deciding factor. Is it possible to implement a way to load IRs directly to TB without such pre-processing?

As for the advantages of TB vs the disadvantages of Netron: could you provide example of how `Loop`/`If` from IR is visualized in TB?

**Comment by rkazants** [comment]
> The _only_ problem with TB is that you must prepare a model to visualize it by writing some code. 
Not sure w

So even if you already have a model represented as a file (IR -- in out case) it is not enough to look at it. If you have the model in Python you still need to serialize it twice -- for IR itself and to format that TB can accept. This is a major and probably deciding factor. Is it possible to implement a way to load IRs directly to TB without such pre-processing?
> 
> As for the advantages of TB vs the disadvantages of Netron: could you provide example of how `Loop`/`If` from IR is visualized in TB?



> The _only_ problem with TB is that you must prepare a model to visualize it by writing some code. So even if you already have a model represented as a file (IR -- in out case) it is not enough to look at it. If you have the model in Python you still need to serialize it twice -- for IR itself and to format that TB can accept. This is a major and probably deciding factor. Is it possible to implement a way to load IRs directly to TB without such pre-processing?
> 
> As for the advantages of TB vs the disadvantages of Netron: could you provide example of how `Loop`/`If` from IR is visualized in TB?



> The _only_ problem with TB is that you must prepare a model to visualize it by writing some code. So even if you already have a model represented as a file (IR -- in out case) it is not enough to look at it. If you have the model in Python you still need to serialize it twice -- for IR itself and to format that TB can accept. This is a major and probably deciding factor. Is it possible to implement a way to load IRs directly to TB without such pre-processing?
> 
> As for the advantages of TB vs the disadvantages of Netron: could you provide example of how `Loop`/`If` from IR is visualized in TB?

Sergey, I will add this support for Loop and If.

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.

**Comment by rkazants** [comment]
No need to close. I hope to continue this effort soon. That is especially needed for huge models.

**Comment by p-wysocki** [comment]
@rkazants there's a `no_stale` label if you need it, it disables the PR closing feature.
